lambdas/tests/unit/services/session-request-validator.test.ts

```diff
@@ -122,11 +127,136 @@ describe("session-request-validator.ts", () => {
             expect(response).toEqual({
                 client_id: "request-client-id",
                 redirect_uri: "redirect-uri",
+                scope: scope,
+                state: state,
                 shared_claims: personIdentity,
             });
         });
     });
 
+    describe("sessionRequestValidator", () => {
+        let sessionRequestValidator: SessionRequestValidator;
+        let sessionRequestValidationConfig: SessionRequestValidationConfig;
+        const jwtVerifier = jest.mocked(JwtVerifier);
+
+        beforeEach(() => {
+            sessionRequestValidationConfig = {
+                expectedJwtRedirectUri: "redirect-uri",
+            } as SessionRequestValidationConfig;
+
+            sessionRequestValidator = new SessionRequestValidator(
+                sessionRequestValidationConfig,
+                jwtVerifier.prototype,
+            );
+        });
+
+        it("should pass when jwt body is correct", async () => {
+            const client_id = "request-client-id";
+            const redirect_uri = "redirect-uri";
+            const scope = "openid";
+            const state = "state";
+
+            const jwtPayload = {
+                client_id: client_id,
+                redirect_uri: redirect_uri,
+                scope: scope,
+                state: state,
+                shared_claims: personIdentity,
+            } as JWTPayload;
+
+            jest.spyOn(jwtVerifier.prototype, "verify").mockReturnValue(
+                new Promise<JWTPayload | null>((res) => res(jwtPayload)),
+            );
+
+            const payload = (await sessionRequestValidator.validateJwt(
+                Buffer.from("test-jwt"),
+                client_id,
+            )) as JWTPayload;
+
+            await expect(payload).toEqual(jwtPayload);
+        });
+
+        it("should fail to validate the jwt if scope is not openid", async () => {
+            const client_id = "request-client-id";
+            const redirect_uri = "redirect-uri";
+            const scope = "test";
+            const state = "state";
+
+            jest.spyOn(jwtVerifier.prototype, "verify").mockReturnValue(
+                new Promise<JWTPayload | null>((res) =>
```

Comment:
This all looks good and functional, could you just make a few clean up instead of new Promise<...> do a Promise.resolve

            jest.spyOn(jwtVerifier.prototype, "verify").mockReturnValueOnce(
                Promise.resolve({
                        client_id
                        redirect_uri
                        scope
                        state
                        shared_claims: personIdentity,
                } as JWTPayload),
            );